packages/curve-ui-kit/src/features/select-token/ui/modal/TokenOption.tsx

```diff
@@ -0,0 +1,61 @@
+import MenuItem from '@mui/material/MenuItem'
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+
+import { TokenIcon } from '@ui-kit/shared/ui/TokenIcon'
+import { formatNumber, FORMAT_OPTIONS } from '@ui/utils'
+
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { IconSize } = SizesAndSpaces
+
+import type { TokenOption as Option } from '../../types'
+
+export type TokenOptionCallbacks = {
+  onToken: () => void
+}
+
+export type TokenOptionsProps = {
+  balance?: string
+  tokenPrice?: number
+  disabled?: boolean
+}
+
+export type Props = Option & TokenOptionCallbacks & TokenOptionsProps
+
+export const TokenOption = ({ chain, symbol, label, address, balance, tokenPrice, disabled, onToken }: Props) => {
+  const hasBalance = +(balance ?? '0') > 0
+  const hasBalanceUsd = hasBalance && (tokenPrice ?? 0 > 0)
+
+  return (
+    <MenuItem
+      onClick={disabled ? undefined : onToken}
+      sx={{
+        minHeight: IconSize.xxl,
+        ...(disabled && {
+          opacity: 0.5,
+          cursor: 'not-allowed',
+        }),
+      }}
+    >
+      {/* I'm abusing the symbol here to deliberately show a tooltip for the address instead */}
```

Comment:
why? :thinking:

Reply:
I've experimented with showing the address on hover and keyboard focus at the place where the label would be, but because we're not using labels it causes the token symbol to move up. See video attached, the symbol typography has to move up to make room for the address to show up. When no longer hovering or focusing the item the address would go away and the symbol would move down again.

This looks very jumpy and distracting so I've opted to keep the token on hover for now

https://github.com/user-attachments/assets/5233f53e-e299-4904-b3e5-6764be2b39cd

Reply:
Renamed the property from 'symbol' to 'tooltip' to more clearly indicate its intent.
As discussed with tutti, we've decided to implement two sections in the modal: "My tokens" and "All tokens"

In "My tokens", we will show balances as we currently do. In "All tokens" we will show the token address (where currently nothing is shown).